src/RwaUrn.sol

```diff
@@ -0,0 +1,305 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
```

Comment:
I'm assuming the only changes are comments?

Reply:
All methods are verbatim copies of the example repo, but the order of the declaration changes.

I also extracted Math related functions into a library with only `internal` functions to improve organization. Because it's an internal library, the compiler inlines it into the contracts using them, so it remains functionally the same as declaring it in the contract.